.github/workflows/unit-tests.yml

```diff
@@ -114,12 +114,27 @@ jobs:
           path: ${{ github.run_id }}
 
       - name: Run tests
+        # Tests in this stage generate code coverage metrics for the repository
+        # Coverage data is uploaded to Codecov in subsequent stages
         env:
           BIONEMO_DATA_SOURCE: ngc
         run: ./ci/scripts/run_pytest.sh --no-nbval --skip-slow
 
+      - name: Run slow tests
+        if: |
+          github.event_name == 'merge_group' ||
+          (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'INCLUDE_SLOW_TESTS'))
+        env:
+          BIONEMO_DATA_SOURCE: ngc
+        run: pytest -m "slow" sub-packages/
+
       - name: Run notebook tests
-        if: ${{ contains(github.event.pull_request.labels.*.name, 'INCLUDE_NOTEBOOKS_TESTS') }}
+        if: |
+          github.event_name == 'merge_group' ||
+          (github.event_name == 'pull_request' && (
+            contains(github.event.pull_request.labels.*.name, 'INCLUDE_NOTEBOOKS_TESTS') ||
+            contains(steps.changed-files.outputs.all_changed_files, '.ipynb')
```

Comment:
@pstjohn , added section to run this test if ipython changes

Reply:
`steps.changed-files.outputs.all_changed_files` did we add this? does it work? I never got my PR to work. I think if we're blocking PRs in the merge queue this is less critical

Reply:
ok!